app/controllers/unaccompanied_controller.rb

```diff
@@ -55,73 +55,71 @@ def start_application
 
   def display
     @application = UnaccompaniedMinor.find_by_reference(session[:app_reference])
+    step = params["stage"].to_i
+    @feature_save_and_return_active = (ENV["UAM_FEATURE_SAVE_AND_RETURN_ACTIVE"] == "true")
 
-    Rails.logger.debug "App JSON: #{@application.as_json}"
+    if @application.nil? || (1..MAX_STEPS).exclude?(step)
+      redirect_to "/sponsor-a-child" and return
+    end
 
-    step = params["stage"].to_i
+    Rails.logger.debug "Step:#{step} App:#{@application.as_json}"
```

Comment:
```suggestion
```

I suggest removing this mostly because we already have a cluttered log and this won't really be needed

Feel free to discard this suggestion, other than this the pr is 👍